### PR TITLE
Resolve substitutions upon SecurityNodeActionExtension._setup_security()

### PR DESCRIPTION
**Pull request by hidmic**
FYI @wjwwood @gbiggs.

### CODE REVIEW

**Review by gbiggs — APPROVED** [review]

File: ros2launch_security/node_action/security.py [hunk 1]
```diff
@@ -50,8 +51,12 @@ def _setup_security(
     ) -> None:
         """Enable encryption, creating a key for the node if necessary."""
         nodl_node = nodl.get_node_by_executable(
-            package_name=node_action.node_package,
-            executable_name=node_action.node_executable
+            package_name=perform_substitutions(
+                context, node_action.node_package
```
Comment (wjwwood) [inline ros2launch_security/node_action/security.py]: This isn't always a list of substitutions (as perform substitutions expects), so you get this sometimes:

```
$ ros2 launch demo_nodes_cpp talker_listener.launch.py --secure
[INFO] [launch]: All log files can be found below /home/william/.ros/log/2021-06-03-13-12-52-151332-ubuntu-1615826
[INFO] [launch]: Default logging verbosity is set to INFO
Task exception was never retrieved
future: <Task finished name='Task-2' coro=<LaunchService._process_one_event() done, defined at /opt/ros/rolling/lib/python3.8/site-packages/launch/launch_service.py:226> exception=AttributeError("'str' object has no attribute 'perform'")>
Traceback (most recent call last):
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/launch_service.py", line 228, in _process_one_event
    await self.__process_event(next_event)
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/launch_service.py", line 248, in __process_event
    visit_all_entities_and_collect_futures(entity, self.__context))
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
    futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
    futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 45, in visit_all_entities_and_collect_futures
    futures_to_return += visit_all_entities_and_collect_futures(sub_entity, context)
  [Previous line repeated 1 more time]
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/utilities/visit_all_entities_and_collect_futures_impl.py", line 38, in visit_all_entities_and_collect_futures
    sub_entities = entity.visit(context)
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/action.py", line 108, in visit
    return self.execute(context)
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch_ros/actions/node.py", line 476, in execute
    cmd_extension, ros_specific_arguments = extension.prepare_for_execute(
  File "/home/william/ros2launch_security_ws/build/ros2launch_security/ros2launch_security/node_action/security.py", line 37, in prepare_for_execute
    ros_specific_arguments = self._setup_security(
  File "/home/william/ros2launch_security_ws/build/ros2launch_security/ros2launch_security/node_action/security.py", line 54, in _setup_security
    package_name=perform_substitutions(
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/utilities/perform_substitutions_impl.py", line 26, in perform_substitutions
    return ''.join([context.perform_substitution(sub) for sub in subs])
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/utilities/perform_substitutions_impl.py", line 26, in <listcomp>
    return ''.join([context.perform_substitution(sub) for sub in subs])
  File "/opt/ros/rolling/lib/python3.8/site-packages/launch/launch_context.py", line 197, in perform_substitution
    return substitution.perform(self)
AttributeError: 'str' object has no attribute 'perform'
```

I think there's a normalize function that ensures everything is a substitution.

Comment (wjwwood) [reply ros2launch_security/node_action/security.py]: I'll work on this now, btw.

Comment (wjwwood) [reply ros2launch_security/node_action/security.py]: @hidmic can you give me access to your account to push (on this pull request branch), or should I make a new pull request with a branch on this repository?

Comment (wjwwood) [reply ros2launch_security/node_action/security.py]: Once I can push, my version of this branch works in combination with https://github.com/ros2/demos/pull/523

Comment (hidmic) [reply ros2launch_security/node_action/security.py]: > I think there's a normalize function that ensures everything is a substitution.

Good catch. Don't know why I assumed the `node_package` property would always be a normalized substitution.

> can you give me access to your account to push

Granted you access to the fork.

**Comment by hidmic** [comment]
Circling back. I think we need extra permissions here, don't why @wjwwood? @arnatious would you mind?

**Comment by wjwwood** [comment]
Yeah, I'll poke people offline too.

**Comment by wjwwood** [comment]
I got permissions now, @Arnatious do you mind if I merge this (and other pull requests) or would you prefer to review and merge pull requests here?

**Comment by wjwwood** [comment]
CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14754/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9504/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12418/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14901/)

**Comment by wjwwood** [comment]
I have to include `nodl` in the CI, I'll fix that shortly.

**Comment by wjwwood** [comment]
New CI with `nodl`:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/14755/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/9505/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/12419/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/14902/)

**Pull request merged**